Mage.Sets/src/mage/cards/j/JabarisInfluence.java

```diff
@@ -0,0 +1,102 @@
+import mage.constants.Duration;
+import mage.constants.WatcherScope;
+import mage.counters.CounterType;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.ObjectSourcePlayer;
+import mage.filter.predicate.ObjectSourcePlayerPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetCreaturePermanent;
+import mage.watchers.Watcher;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * @author Cguy7777
+ */
+public final class JabarisInfluence extends CardImpl {
+
+    private static final FilterCreaturePermanent filter = new FilterCreaturePermanent("nonartifact, nonblack creature that attacked you this turn");
+
+    static {
+        filter.add(JabarisInfluencePredicate.instance);
+    }
+
+    public JabarisInfluence(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.INSTANT}, "{3}{W}{W}");
+
+        // Cast Jabari's Influence only after combat.
+        this.addAbility(new CastOnlyDuringPhaseStepSourceAbility(
+                null, null, AfterCombatCondition.instance, "Cast this spell only after combat"));
+
+        // Gain control of target nonartifact, nonblack creature that attacked you this turn and put a -1/-0 counter on it.
+        this.getSpellAbility().addEffect(new GainControlTargetEffect(Duration.Custom));
+        this.getSpellAbility().addEffect(new AddCountersTargetEffect(CounterType.M1M0.createInstance()).setText("and put a -1/-0 counter on it"));
+        this.getSpellAbility().addTarget(new TargetCreaturePermanent(filter));
+        this.getSpellAbility().addWatcher(new JabarisInfluenceWatcher());
+    }
+
+    private JabarisInfluence(final JabarisInfluence card) {
+        super(card);
+    }
+
+    @Override
+    public JabarisInfluence copy() {
+        return new JabarisInfluence(this);
+    }
+}
+
+enum JabarisInfluencePredicate implements ObjectSourcePlayerPredicate<Permanent> {
+    instance;
+
+    @Override
+    public boolean apply(ObjectSourcePlayer<Permanent> input, Game game) {
+        Permanent permanent = input.getObject();
+
+        JabarisInfluenceWatcher watcher = game.getState().getWatcher(JabarisInfluenceWatcher.class);
+        return watcher.checkIfEligibleCreatureAttackedYouThisTurn(permanent, input.getPlayerId(), game);
+    }
+}
+
+class JabarisInfluenceWatcher extends Watcher {
+
+    private final Map<MageObjectReference, UUID> eligibleAttackedThisTurnCreatures = new HashMap<>();
+
+    public JabarisInfluenceWatcher() {
+        super(WatcherScope.GAME);
+    }
+
+    @Override
+    public void watch(GameEvent event, Game game) {
+        if (event.getType() == GameEvent.EventType.ATTACKER_DECLARED) {
+            Permanent permanent = game.getPermanent(event.getSourceId());
+            if (permanent != null && !permanent.isArtifact(game) && !permanent.getColor(game).equals(ObjectColor.BLACK)) {
```

Comment:
I don't think this is correct on a technicality.

My interpretation of the card wording is that the object being targeted must not be an artifact, must not be black, and must have attacked you this turn. I do not believe it is required for the object to have been nonartifact and nonblack at the time it attacked.

Thus I would recommend the watcher only deals with the attacking you, and use existing predicates for the type/color.

Reply:
I asked on the [Live Rules Q&A](https://chat.magicjudges.org/mtgrules/) to be certain, and you are correct. My bad! I've changed it so that it will only check if the creature is nonartifact/nonblack on targeting (and resolution).